classes/03class/exercises/c03-aws01/alanlima/README.md

```diff
@@ -0,0 +1,210 @@
+# C03-aws01
+
+## Attachments
+
+- [User Data Script](user-data.txt)
+
+## Command Execution Output
+
+- Commands used to create your autoscaling group:
+
+    ```bash
+    # Create a new Security group to allow connection to port 80
+    $ aws ec2 create-security-group \
+        --group-name c03-aws01-sg \
+        --description "Allow connection to port 80" \
+        --vpc-id vpc-08fb60055e82efe81 \
+        --tag-specifications "ResourceType=security-group,Tags=[{Key=Name,Value=c03-aws01-sg},{Key=DevopsClass,Value=c03}]"
+
+    GroupId: sg-07739f756a518e82e
+    Tags:
+    - Key: Name
+    Value: c03-aws01-sg
+    - Key: DevopsClass
+    Value: c03
+
+    # Update the security group to allow ingress from anywhere to port 80
+    $ aws ec2 authorize-security-group-ingress \
+        --group-id sg-07739f756a518e82e \
+        --cidr 0.0.0.0/0 \
+        --port 80 \
+        --protocol tcp
+
+    # Create the Launch Configuration resource with the template details to be applied for each instance when created
+    $ aws autoscaling create-launch-configuration \
+        --launch-configuration-name devops-init-instance-data \
+        --user-data file://user-data.txt \
+        --image-id ami-088ff0e3bde7b3fdf \
+        --instance-type t2.micro \
+        --security-groups sg-07739f756a518e82e
```

Comment:
Here you didn't specify ` --associate-public-ip-address` and that's alright.

The default behavior is `Only assign a public IP address to instances launched in a subnet with auto-assign public IP enabled (default)`

If you were using subnets without auto-assign public IP enabled, this would be a problem.

Reply:
I had that option enabled for the public subnet. I added to the launch configuration as well to avoid problems 😅